components/style/values/computed/basic_shape.rs

```diff
@@ -10,10 +10,43 @@
 use cssparser::ToCss;
 use properties::shorthands::serialize_four_sides;
 use std::fmt;
+use url::Url;
 use values::computed::position::Position;
 use values::computed::{BorderRadiusSize, LengthOrPercentage};
+use values::computed::UrlExtraData;
 
-pub use values::specified::basic_shape::FillRule;
+pub use values::specified::basic_shape::{FillRule, GeometryBox, ShapeBox};
+
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+pub enum ShapeSource<T> {
```

Comment:
No, the types are different -- computed::ShapeSource contains a computed::BasicShape, which isn't the same as a specified::BasicShape